File: 3408-count-the-number-of-special-characters-i/solution.java
class Solution {
    public int numberOfSpecialChars(String word) {
        Map<Character, Character> map = new HashMap<>();

        int n = word.length();
        int count = 0;

        for(int i=0; i<n; i++) {
            char ch = word.charAt(i);
            if(ch >= 'a' && ch <= 'z') {
                map.put(ch, Character.toUpperCase(ch));
            }
        }
        //a-A,b-B,c-C
        for(Character k: map.keySet()) {
            System.out.println(k + " " + map.get(k));
        }
        for(int i=0; i<n; i++) {
            char ch = word.charAt(i);
            if(ch >= 'A' && ch <= 'Z') {
                if(map.containsKey(Character.toLowerCase(ch))) {
                    count ++;
                    map.remove(Character.toLowerCase(ch));
                }
            }
        }
        return count;
    }
}

File: 3408-count-the-number-of-special-characters-i/solution.swift
class Solution {
    func numberOfSpecialChars(_ word: String) -> Int {
        var lowerSet = Set<Character>()
        var upperSet = Set<Character>()

        
        for ch in word {
            // if small letters
            if ch.isLowercase {
                lowerSet.insert(ch)
            } else {
                upperSet.insert(Character(ch.lowercased()))
            }
        }

        // print(lowerSet, "\n")//["a", "c", "b"] 
        // print(upperSet)//["c", "a", "b"]

        let count = lowerSet.intersection(upperSet).count
        return count
    }
}
